src/history/hs.c

```diff
@@ -307,9 +307,27 @@ static int
 __hs_row_search(WT_CURSOR_BTREE *hs_cbt, WT_ITEM *srch_key, bool insert)
 {
     WT_DECL_RET;
+    bool leaf_found;
+
+    leaf_found = false;
+
+    /*
+     * Check whether the search key can be find in the provided leaf page, if exists. Otherwise
+     * perform a full search.
+     */
+    if (hs_cbt->ref != NULL)
+        WT_WITH_BTREE(CUR2S(hs_cbt), CUR2BT(hs_cbt),
+          ret = __wt_row_search(hs_cbt, srch_key, insert, hs_cbt->ref, false, &leaf_found));
```

Comment:
If we get a non-zero return here, it'll potentially get ignored overwritten by the next search.
I'm guessing `WT_NOTFOUND` might be expected but other values, probably not.

Reply:
Thanks. I guess the earlier call is not expecting a WT_NOTFOUND also. As we are trying to insert a key, I am not sure whether we need to check for WT_NOTFOUND. I feel adding WT_RET(ret) should work. Am I missing anything?

Reply:
Yep. It looks like even if we don't find it, we set `leaf_found` to `false` and return 0.
I agree with you.

Reply:
Fixed with WT_RET. Thanks.